src/Hal/Component/Ast/NodeTraverser.php

```diff
@@ -27,11 +27,7 @@ public function __construct($cloneNodes = false, $stopCondition = null)
 
         if(null === $stopCondition) {
             $stopCondition = function($node) {
-                if($node instanceof Node\Stmt\Class_ || $node instanceof Node\Stmt\Interface_) {
-                    return false;
-                }
-
-                return true;
+                return !$node instanceof Node\Stmt\Class_ || !$node instanceof Node\Stmt\Interface_;
```

Comment:
I guess you meant `&&` instead of `||`

Reply:
Yes, reverted the logic, but didn't revert the operands :sweat_smile: